packages/react-inbox/src/hooks/use-inbox.ts

```diff
@@ -1,47 +1,144 @@
+  const [_, trackEvent] = useTrackEvent();
+
+  const newMessage = (payload) => {
+    dispatch({
+      type: "inbox/NEW_MESSAGE",
+      payload,
+    });
+  };
+
+  const setLoading = (isLoading: boolean) => {
+    dispatch({
+      type: "inbox/SET_LOADING",
+      payload: {
+        isLoading,
+      },
+    });
+  }
+
+  useEffect(() => {
+    transport?.listen({
+      id: "inbox-listener",
+      listener: (courierEvent) => {
+        newMessage(courierEvent?.data);
+      },
+    });
+  }, [transport]);
 
   return {
     ...inbox,
+
+    setLoading,
     init: (payload) => {
       dispatch({
         type: "inbox/INIT",
-        payload,
+        payload: {
+          ...payload,
+          tabs: payload.tabs ?? [{
+            id: "unread",
+            label: "Unread",
+            filter: {
+              isRead: false
+            }
+          }, {
+            id: "all",
+            label: "All Messages",
+            filter: {}
+          }]
+        }
       });
     },
 
-    setLoading: (isLoading: boolean) => {
+    setCurrentTab: (newTab: string) => {
       dispatch({
-        type: "inbox/SET_LOADING",
-        payload: {
-          isLoading,
-        },
+        type: "inbox/SET_CURRENT_TAB",
+        payload: newTab
       });
     },
 
-    setHasUnreadMessages: (hasUnreadMessages: boolean) => {
+    fetchMessages: async (params: {
+      params?: {
+        after?: string;
+        isRead?: boolean;
+      }
+    }) => {
       dispatch({
-        type: "inbox/SET_HAS_UNREAD_MESSAGES",
+        type: "inbox/FETCH_MESSAGES",
+        payload: () => fetchMessages(params)
+      });
+    },
+
+    setHasUnreadMessages: (unreadMessageCount: number) => {
+      dispatch({
+        type: "inbox/SET_UNREAD_MESSAGE_COUNT",
         payload: {
-          hasUnreadMessages,
+          unreadMessageCount,
         },
       });
     },
 
+    setMessages: (payload) => {
+      dispatch({
+        type: "inbox/SET_MESSAGES",
+        payload,
+      })
+    },
+
     addMessages: (payload) => {
       dispatch({
         type: "inbox/ADD_MESSAGES",
         payload,
+      })
+    },
+
+    setUnreadMessages: (payload) => {
+      dispatch({
+        type: "inbox/SET_UNREAD_MESSAGES",
+        payload
       });
     },
 
-    newMessage: (payload) => {
+    setAllMessages: (payload) => {
       dispatch({
-        type: "inbox/NEW_MESSAGE",
-        payload,
+        type: "inbox/SET_ALL_MESSAGES",
+        payload
+      });
+    },
+
+    markMessageRead: async (messageId: string, trackingId: string) => {
+      await trackEvent({
+        trackingId
+      });
+
+      dispatch({
+        type: "inbox/MARK_MESSAGE_READ",
+        payload: {
+          messageId
+        }
       });
     },
+
+    markMessageUnread: async (messageId: string, trackingId: string) => {
+      await trackEvent({
+        trackingId
+      });
+
+      dispatch({
+        type: "inbox/MARK_MESSAGE_UNREAD",
```

Comment:
Would like to have this as a constant namespace in the future

Reply:
agree